app/_components/pdf-viewer.tsx

```diff
@@ -59,14 +59,14 @@ export default function PdfViewer({
   function onDocumentLoadSuccess(docProxy: PDFDocumentProxy) {
     const numPages = docProxy.numPages;
     setPageCount(numPages);
-    setLayers(new Map());
     setPageSize({ action: "clear" });
     if (stitchSettings.pageRange.endsWith("-") && numPages > 0) {
       dispatchStitchSettings({
         type: "set-page-range",
         pageRange: stitchSettings.pageRange + numPages,
       });
     }
+    getLayersFromPdf(docProxy).then((l) => setLayers(l));
```

Comment:
It felt nicer to do this here, rather than first setting an empty map and then setting the layers during rendering.